src/pybind/mgr/prometheus/module.py

```diff
@@ -819,6 +842,26 @@ def _setup_static_metrics(self) -> Dict[str, Metric]:
 
         return metrics
 
+    @profile_method()
+    def get_osdmap_stats(self) -> None:
+        # retrieve osdmap and epoch clean metrics
+        _, report_raw, _ = self.mon_command({
+            'prefix': 'report',
+            'format': 'json'
+        })
+        try:
+            report = json.loads(report_raw)
+            for stat in OSDMAP_STATS:
+                path = 'osdmap_{}'.format(stat)
+                self.metrics[path].set(report[path])
+            self.metrics['min_last_epoch_clean'].set(
+                report['osdmap_clean_epoch']['min_last_epoch_clean']
+            )
+            for pool in report['osdmap_clean_epochs']['last_epoch_clean']['per_pool']:
+                self.metrics['last_epoch_clean_floor'].set(pool['floor'], (pool['poolid'],))
+        except (json.decoder.JSONDecodeError, KeyError):
+            return
```

Comment:
Do you know if these metrics are provided by any of the existing [ceph-mgr API](https://docs.ceph.com/en/latest/mgr/modules/)? I'm asking that because ceph-mgr provides caching, while here we have to actively query the mgr module about this info.

If you want to explore the API, the mgr now features a [new module for inspecting the mgr API](https://docs.ceph.com/en/latest/mgr/cli_api/) right from the CLI:
```
# ceph mgr module enable cli_api
```

I had a look at the code and regarding these 2 new metrics:
* `last_epoch_clean` is available (per pool) in `get('osd_map')`:
```
# ceph mgr cli get osd_map | grep last_epoch_clean
        "last_epoch_clean": 0,
        "last_epoch_clean": 0,
        "last_epoch_clean": 0,
...
```
* `min_last_epoch_clean` is not directly available, but if I'm not wrong (`mon/OSDMonitor.cc`) is simply the min of the above values, so it could even be calculated as part of a prometheus query.
* `osdmap_first_committed` and `osdmap_last_committed`: if we really want these, we should add a new mgr API call to expose the `OSDMonitor::dump_info()` (or add it under the existing `get(osdmap)`or `get(osd_stats)`).